### DISCUSSION TITLE
Migrating collection name

### DISCUSSION THREAD

**Discussion by lukepighetti** [pocketbase/pocketbase#5095] [Q&A]
If I have a collection "users" and I want to migrate it to "app1_users", it's not really clear to me the best way to achieve this. I considered making a table for name migrations / aliases, but it would be interesting if there was a feature like this built in.

Curious if anyone has an opinion on the best way to achieve this today, and if this is something pocketbase could improve in the future?

**Comment by ganigeorgiev** [comment]
You can apply collection changes programmatically using [Go](https://pocketbase.io/docs/go-collections/#update-existing-collection) or the [JSVM](https://pocketbase.io/docs/js-collections/#update-existing-collection).

Ideally such changes will be executed as part of a migration (again [Go](https://pocketbase.io/docs/go-migrations/) or [JSVM](https://pocketbase.io/docs/js-migrations/)). Unapplied migrations are executed automatically on `serve`.

There is no builtin functionality at the moment for collection aliases. The only use case I can think for this is if you are working on a desktop or mobile app and/or don't have direct control on what versions of your app is being used but it could get very complicated if we also have to take field names in consideration so for now remains out of the scope.

A workaround could be to register a redirect in a middleware or reverse-proxy level that will rewrite the api url to use your new collection.

**Reply by lukepighetti** [reply]
just to note: use case is mobile app + direct collection access via the SDK!
